vwifi.c

```diff
@@ -153,103 +169,30 @@ static inline s32 rand_int_smooth(s32 low, s32 up, s32 seed)
-    }
-    return h;
-}
-
-/* Helper function for generating BSSID from SSID */
-static void generate_bssid_with_ssid(u8 *result, const char *ssid)
-{
-    u64_to_ether_addr(murmurhash(ssid), result);
-    result[0] &= 0xfe; /* clear multicast bit */
-    result[0] |= 0x02; /* set local assignment bit */
-}
-
-/* Update AP database from module parameter ssid_list */
-static void update_ssids(const char *ssid_list)
-{
-    struct ap_info_entry_t *ap;
-    const char delims[] = "[]";
-    struct hlist_node *tmp;
-
-    for (char *s = (char *) ssid_list; *s != 0; /*empty*/) {
-        bool ssid_exist = false;
-        char token[IEEE80211_MAX_SSID_LEN] = {0};
-
-        /* Get the number of token separator characters. */
-        size_t n = strspn(s, delims);
-        s += n; /* Actually skip the separators */
-        /* Get the number of token (non-separator) characters. */
-        n = strcspn(s, delims);
-        if (n == 0)  // token not found
-            continue;
-        strncpy(token, s, n);
-        s += n; /* Point the next token */
-
-        /* Insert the SSID into hash */
-        token[n] = '\0';
-        u32 key = murmurhash((char *) token);
-        hash_for_each_possible_safe (ssid_table, ap, tmp, node, key) {
-            if (strncmp(token, ap->ssid, n) == 0) {
-                ssid_exist = true;
-                break;
-            }
-        }
-        if (ssid_exist) /* SSID exists */
-            continue;
-
-        ap = kzalloc(sizeof(struct ap_info_entry_t), GFP_KERNEL);
-        if (!ap) {
-            pr_err("Failed to alloc ap_info_entry_t incomming SSID=%s\n",
-                   token);
-            return;
-        }
-        u8 bssid[ETH_ALEN] = {0};
-        strncpy(ap->ssid, token, n);
-        generate_bssid_with_ssid(bssid, token);
-        memcpy(ap->bssid, bssid, ETH_ALEN);
-        hash_add(ssid_table, &ap->node, key);
-    }
-}
-
 /* Helper function that will prepare structure with self-defined BSS information
  * and "inform" the kernel about "new" BSS Most of the code are copied from the
  * upcoming inform_dummy_bss function.
  */
-static void inform_dummy_bss(struct owl_vif *vif)
+static void inform_bss(struct owl_vif *vif)
 {
-    struct ap_info_entry_t *ap;
-    int i;
-    struct hlist_node *tmp;
-
-    update_ssids(ssid_list);
-    if (hash_empty(ssid_table))
-        return;
+    struct owl_vif *ap;
 
-    hash_for_each_safe (ssid_table, i, tmp, ap, node) {
+    list_for_each_entry (ap, &owl->ap_list, ap_list) {
         struct cfg80211_bss *bss = NULL;
         struct cfg80211_inform_bss data = {
             /* the only channel */
-            .chan = &vif->wdev.wiphy->bands[NL80211_BAND_2GHZ]->channels[0],
+            .chan = &ap->wdev.wiphy->bands[NL80211_BAND_2GHZ]->channels[0],
             .scan_width = NL80211_BSS_CHAN_WIDTH_20,
-            .signal = rand_int_smooth(-100, -30, jiffies) * 100,
-        };
+            .signal = DBM_TO_MBM(rand_int_smooth(-100, -30, jiffies))};
```

Comment:
Add extra comma (`,`) right after `)`, so that `clang-format` can indent better.

Reply:
OK.